pysipp/launch.py

```diff
@@ -178,3 +180,10 @@ async def finalize(runner, agents, timeout):
         raise SIPpFailure(msg)
 
     return cmds2procs
+
+
+def finalizer(runner, agents):
```

Comment:
the API is cleaner and less error prone as the runner is not exposed to the user anymore which means fewer chances of reaching invalid states. Functional as in less object oriented.

Reply:
> the API is cleaner and less error prone as the runner is not exposed to the user anymore which means fewer chances of reaching invalid states.

Not sure how that's possible to reach *invalid states* but yeah I guess in theory?

If we don't have a *runner* and only a simple function there there's also no future option for adding *cancellation* support, which is kinda what `trio` is all about. I don't want to over-design now or anything but making this the api is yes maybe *simpler* (as much as `.__call__()` vs. `.finalize()` is) but it may be somewhat limiting down the road?

> Functional as in less object oriented.

Because we aren't returning a runner?
Calling `TrioRunner.finalize()` vs. `lambda.__call__()` i'm not really seeing as being any more or less *objecty* in python but 🤷🏼. I still don't understand why you need this function block, why not just return the partial on `trio.run()` if you wanted this; heck it could be a `lambda`?